create-plugin-update/pm.sh

```diff
@@ -15,36 +15,28 @@ install_pnpm_if_not_present() {
 }
 
 # Detect the package manager
+# to determine the correct installation command
+# and the exec command to run create-plugin update
 if [ -f yarn.lock ]; then
-	pm="yarn"
+	pmi=("yarn" "install")
+	pmu=("yarn" "create" "@grafana/plugin" "update")
 elif [ -f pnpm-lock.yaml ]; then
 	install_pnpm_if_not_present
-	pm="pnpm"
+	pmi=("pnpm" "install" "--no-frozen-lockfile")
+	pmu=("pnpm" "dlx" "@grafana/create-plugin@latest" "update")
 elif [ -f package-lock.json ]; then
-	pm="npm"
+	pmi=("npm" "install")
+	pmu=("npx" "-y" "@grafana/create-plugin@latest" "update")
 else
 	echo "No recognized package manager found in this project."
 	exit 1
 fi
 
-# Detect the exec command to run create-plugin update
-if [ -f yarn.lock ]; then
-  pmx=("yarn" "create" "@grafana/plugin")
-elif [ -f pnpm-lock.yaml ]; then
-  install_pnpm_if_not_present
-  pmx=("pnpm" "dlx" "@grafana/create-plugin@latest")
-elif [ -f package-lock.json ]; then
-  pmx=("npx" "-y" "@grafana/create-plugin@latest")
-else
-  echo "No recognized package manager found in this project."
-  exit 1
-fi
-
 # Run the provided command with the detected package manager
 if [ "$1" = "install" ]; then
-  echo "Running '$1' with $pm..."
-	"$pm" install
+	echo "Running '$1' with ${pmi[0]}..."
+	"${pmi[@]}"
 elif [ "$1" = "update" ]; then
-  echo "Running '$1' with ${pmx[0]}..."
-	"${pmx[@]}" update
+	echo "Running '$1' with ${pmu[0]}..."
+	"${pmu[@]}"
```

Comment:
nit: rename this to pm_update_create_plugin

currently this is behind a cryptic name that's not easy to know. one might think this is short for "package manager update" but it is actually "create plugin update"